datasquare_test.go

```diff
@@ -53,18 +53,20 @@ func TestInvalidDataSquareCreation(t *testing.T) {
 
 func TestSetCell(t *testing.T) {
 	ds, err := newDataSquare([][]byte{{1}, {2}, {3}, {4}}, NewDefaultTree)
-	if err != nil {
-		panic(err)
-	}
+	assert.NoError(t, err)
 
-	// SetCell can only write to nil cells
-	assert.Panics(t, func() { ds.SetCell(0, 0, []byte{0}) })
+	err = ds.setCell(0, 0, []byte{42})
+	assert.NoError(t, err)
+	assert.Equal(t, []byte{42}, ds.GetCell(0, 0))
```

Comment:
I think it is worth adding a test for `setCell` to add a cell with the wrong size.